### DISCUSSION TITLE
IsCheckedChanged

### DISCUSSION THREAD

**Discussion by vladrevers** [AvaloniaUI/Avalonia#13283] [General]
Now the Checked observer is marked as obsolete in the JB Rider, and it is proposed to replace it with IsCheckedChanged (which combines Checked and Unchecked). For checkboxes this works fine, no need to write two parameters, but for radio buttons it leads to two consecutive triggers (because one radio button becomes Checked and the other Unchecked.. ). Is this a issue or is there a way around it?

Upd: The last sender is RadioButton which became "Unchecked", which can also cause a problem if you just replace the Checked parameter with IsCheckedChanged.

No workarounds like:
if (((RadioButton) sender).IsChecked != true) return;

**Comment by thevortexcloud** [comment]
If you only want it to trigger on checked or unchecked you don't have a lot of other options besides checking the current value.

**Reply by robloo** [reply]
IsCheckedChanged is a simplification to the API carried over from WinUI's newer controls. We had a chance to apply it to older controls too to make the API uniform.

In this case, it is correct to read the value of the IsChecked property directly. I would recommend:

```csharp
if (sender is RadioButton radioButton)
{
    if (radioButton.IsChecked)
    {
        // Checked
    }
    else
    {
        // Unchecked
    }
}
```
